### PR TITLE
chore: add test retries

### PR DESCRIPTION
**Pull request by mxschmitt**
This PR adds the basic infra to implement retries inside the `Microsoft.Playwright.NUnit` and `Microsoft.Playwright.MSTest` harnesses. It's done via [C# attributes](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/) which override the execution logic and re-runs it until the max-retries is reached or the test has passed.

~~The `Test2RetryCount` data structure is a naive way of sharing the retry-count with the actual test execution (by full classname/method name as a key), which is required for implementing screenshot/video/tracing on-first-retry (follow-up work)~~. We now use Test Properties for it.

Follow-up work related to retries implementation:

- Improve the retry reporting, we should print inside the test log the logs of the previous failed test runs, stdout, stderr, exceptions, timeouts?. Right now only the amount of tests gets printed.
- Do not require `[Test]` and `[PlaywrightTest]` for NUnit, it looks like we run into this bug: https://github.com/nunit/nunit/issues/2316 or https://github.com/nunit/nunit/issues/1347 so the users needs to either add `[Test]` and `[PlaywrightTest]` to their test or `[TestFixture]` to their class to workaround the discovery issue.

Relates https://github.com/microsoft/playwright-dotnet/issues/2214

<details>
  <summary>Example MSTest output</summary>
  
  ```log
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
/Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/bin/Debug/net6.0/Playwright.TestingHarnessTest.dll
NUnit Adapter 4.2.0.0: Test execution started
Running selected tests in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/bin/Debug/net6.0/Playwright.TestingHarnessTest.dll
   NUnit3TestExecutor discovered 0 of 0 NUnit test cases using Current Discovery mode, Explicit run
NUnit Adapter 4.2.0.0: Test execution complete
  Passed Test [615 ms]
  Standard Output Messages:
 i-was-running-stdout4
 
 ================================================================================
 Test was retried 4 times.
 Failing test runs:
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed (retry #0)
   Outcome: Failed
   Exception: 
     Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel.TestFailedException: Test method Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test threw exception: 
     System.Exception: i-was-broken
      ---> System.Exception: i-was-broken
        at Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test() in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/ExampleTests.cs:line 19
        --- End of inner exception stack trace ---
   Standard Output Messages: 
     i-was-running-stdout0
   Standard Error Messages: 
     i-was-running-stderr0
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed (retry #1)
   Outcome: Failed
   Exception: 
     Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel.TestFailedException: Test method Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test threw exception: 
     System.Exception: i-was-broken
      ---> System.Exception: i-was-broken
        at Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test() in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/ExampleTests.cs:line 19
        --- End of inner exception stack trace ---
   Standard Output Messages: 
     i-was-running-stdout1
   Standard Error Messages: 
     i-was-running-stderr1
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed (retry #2)
   Outcome: Failed
   Exception: 
     Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel.TestFailedException: Test method Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test threw exception: 
     System.Exception: i-was-broken
      ---> System.Exception: i-was-broken
        at Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test() in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/ExampleTests.cs:line 19
        --- End of inner exception stack trace ---
   Standard Output Messages: 
     i-was-running-stdout2
   Standard Error Messages: 
     i-was-running-stderr2
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed (retry #3)
   Outcome: Failed
   Exception: 
     Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel.TestFailedException: Test method Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test threw exception: 
     System.Exception: i-was-broken
      ---> System.Exception: i-was-broken
        at Playwright.TestingHarnessTest.MSTest.mstestretryspectsshouldretryafailedtestandstoponceitpassed.Test() in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/ExampleTests.cs:line 19
        --- End of inner exception stack trace ---
   Standard Output Messages: 
     i-was-running-stdout3
   Standard Error Messages: 
     i-was-running-stderr3
 ================================================================================
  Standard Error Messages:
 i-was-running-stderr4

Html test results file : /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/TestResults/testResults.html

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 6.5994 Seconds

  ```
</details>

<details>
  <summary>Example NUnit output</summary>
  
  ```log
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
/Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/bin/Debug/net6.0/Playwright.TestingHarnessTest.dll
NUnit Adapter 4.2.0.0: Test execution started
Running selected tests in /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/bin/Debug/net6.0/Playwright.TestingHarnessTest.dll
   NUnit3TestExecutor discovered 1 of 1 NUnit test cases using Current Discovery mode, Non-Explicit run
i-was-running-stderr0
i-was-running-stderr1
i-was-running-stderr2
i-was-running-stderr3
i-was-running-stderr4
i-was-running-stdout4

================================================================================
Test was retried 4 times.
Failing test runs:
----------------------------------------
  Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
  Outcome: Failed:Error
  Standard Output Messages: 
    i-was-running-stdout1
  Message 
    System.Exception : i-was-broken
----------------------------------------
  Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
  Outcome: Failed:Error
  Standard Output Messages: 
    i-was-running-stdout2
  Message 
    System.Exception : i-was-broken
----------------------------------------
  Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
  Outcome: Failed:Error
  Standard Output Messages: 
    i-was-running-stdout3
  Message 
    System.Exception : i-was-broken
----------------------------------------
  Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
  Outcome: Passed
  Standard Output Messages: 
    i-was-running-stdout4
================================================================================
NUnit Adapter 4.2.0.0: Test execution complete
  Passed Test [2 s]
  Standard Output Messages:
 i-was-running-stdout4
 
 ================================================================================
 Test was retried 4 times.
 Failing test runs:
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
   Outcome: Failed:Error
   Standard Output Messages: 
     i-was-running-stdout1
   Message 
     System.Exception : i-was-broken
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
   Outcome: Failed:Error
   Standard Output Messages: 
     i-was-running-stdout2
   Message 
     System.Exception : i-was-broken
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
   Outcome: Failed:Error
   Standard Output Messages: 
     i-was-running-stdout3
   Message 
     System.Exception : i-was-broken
 ----------------------------------------
   Test: Playwright.TestingHarnessTest.NUnit.nunitretryspectsshouldretryafailedtestandstoponceitpassed.Test
   Outcome: Passed
   Standard Output Messages: 
     i-was-running-stdout4
 ================================================================================
  Standard Error Messages:
 i-was-running-stderr0
 i-was-running-stderr1
 i-was-running-stderr2
 i-was-running-stderr3
 i-was-running-stderr4

Html test results file : /Users/max/development/playwright-dotnet/src/Playwright.TestingHarnessTest/TestResults/testResults.html

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 3.4741 Seconds

  ```
</details>

### CODE REVIEW

File: src/Playwright.TestAdapter/TestHarnessStorage.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Playwright.TestAdapter
+{
+    public class TestHarnessStorage
+    {
+        public static readonly IDictionary<string, int> _runCountPerTest = new Dictionary<string, int>();
```
Comment (kblok) [inline src/Playwright.TestAdapter/TestHarnessStorage.cs:32]: If it's public I would expose it as a read-only property.

File: src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            while (!TestHarnessStorage.IsLastRun(key))
+            {
+                var results = base.Execute(testMethod);
+                var allPassed = (results ?? new MSTestUnitTesting.TestResult[] { }).All(r => r.Outcome == MSTestUnitTesting.UnitTestOutcome.Passed);
+
+                if (results != null && TestHarnessStorage._runCountPerTest[key] > 0)
+                {
+                    foreach (var result in results)
+                    {
+                        result.DisplayName = testMethod.TestMethodName + " (retry #" + TestHarnessStorage._runCountPerTest[key] + ")";
```
Comment (kblok) [inline src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs:48]:
```suggestion
                        result.DisplayName = $"{testMethod.TestMethodName} (retry #{TestHarnessStorage._runCountPerTest[key]})";
```

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        }
+
+        public TestCommand Wrap(TestCommand command)
+            => new RetryTestCommand(command);
+
+        public class RetryTestCommand : DelegatingTestCommand
+        {
+            public RetryTestCommand(TestCommand innerCommand)
+                : base(innerCommand)
```
Comment (kblok) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:48]:
```suggestion
            public RetryTestCommand(TestCommand innerCommand) : base(innerCommand)
```

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            {
+            }
+
+            public override TestResult Execute(TestExecutionContext context)
+            {
+                Console.WriteLine("Execute" + PlaywrightSettingsProvider.Retries);
```
Comment (kblok) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:54]: Do we need this?

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                string key = Test.Id;
+                while (!TestHarnessStorage.IsLastRun(key))
+                {
+                    try
+                    {
+                        context.CurrentResult = innerCommand.Execute(context);
+                    }
+                    catch (System.Exception ex)
```
Comment (kblok) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:62]:
```suggestion
                    catch (Exception ex)
```

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    {
+                        context.CurrentResult = context.CurrentTest.MakeTestResult();
+                        context.CurrentResult.RecordException(ex);
+                    }
+                    TestHarnessStorage.IncrementRunCount(key);
+
+                    if (context.CurrentResult.ResultState == ResultState.Success)
+                        break;
+                    if (context.CurrentResult.ResultState == ResultState.Failure && TestHarnessStorage.IsLastRun(key))
+                        break;
```
Comment (kblok) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:72]:
```suggestion
                    if (context.CurrentResult.ResultState == ResultState.Success)
                    {
                        break;
                    }
                    if (context.CurrentResult.ResultState == ResultState.Failure && TestHarnessStorage.IsLastRun(key))
                    {
                        break;
                    }
```

File: src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs [hunk 7]
```diff
@@ -73,6 +73,15 @@ public static string BrowserName
             private set { }
         }
 
+        public static int Retries
+        {
+            get
+            {
+                return _settings?.Retries ?? 0;
+            }
```
Comment (kblok) [inline src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs:81]:
```suggestion
            get => _settings?.Retries ?? 0;
```

File: src/Playwright.TestAdapter/TestHarnessStorage.cs (hunk 1 above)
Comment (mxschmitt) [reply src/Playwright.TestAdapter/TestHarnessStorage.cs:32]: I think I'll move it out from the TestAdapter package into the NUnit and MSTest package which is then internal.

File: src/Playwright.TestingHarnessTest/tests/mstest/retry.spec.ts [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+      }`,
+      '.runsettings': `
+      <?xml version="1.0" encoding="utf-8"?>
+      <RunSettings>
+        <Playwright>
+          <Retries>10</Retries>
+        </Playwright>
+      </RunSettings>`,
+  }, 'dotnet test --settings=.runsettings');
+  expect(result.exitCode).toBe(1);
```
Comment (mxschmitt) [inline src/Playwright.TestingHarnessTest/tests/mstest/retry.spec.ts:184]: note: ask mstest people if we can intercept the exitCode logic.

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        public class RetryTestCommand : DelegatingTestCommand
+        {
+            public RetryTestCommand(TestCommand innerCommand) : base(innerCommand)
+            {
+            }
+
+            public override TestResult Execute(TestExecutionContext context)
+            {
+                string key = Test.Id;
+                while (!TestHarnessStorage.IsLastRun(key))
```
Comment (mxschmitt) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:53]: note: don't report the intermediate test results and print the log to the output of the result.

File: src/Playwright.MSTest/PlaywrightTest.cs [hunk 10]
```diff
@@ -121,10 +121,11 @@ protected bool TestOK
                 || TestContext!.CurrentTestOutcome == UnitTestOutcome.NotRunnable;
         }
 
-        public TestContext? TestContext { get; set; }
+        public TestContext TestContext { get; set; } = null!;
```
Comment (mxschmitt) [inline src/Playwright.MSTest/PlaywrightTest.cs:124]: drive-by fix.

File: src/Playwright.TestingHarnessTest/tests/baseTest.ts [hunk 11]
```diff
@@ -117,4 +117,14 @@ function unintentFile(content: string): string {
   return lines.join('\n').trim();
 }
 
+// https://github.com/microsoft/vstest/blob/200a783858425e5ac6f4ebb8f87a7811b0ad39e3/src/vstest.console/Internal/ConsoleLogger.cs#L319-L343
+function extractVstestMessages(stdout: string, prefix: string): string {
```
Comment (mxschmitt) [inline src/Playwright.TestingHarnessTest/tests/baseTest.ts:121]: nunit/mstest or more in general the visual studio test platform will report the stdout/stderr after each test run with a prefix and two blank lines as a end. This parses them so we can easier process them. This seemed easier to trx, but can be changed most likely to get it from trx instead.

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+        public class RetryTestCommand : DelegatingTestCommand
+        {
+            public RetryTestCommand(TestCommand innerCommand) : base(innerCommand)
+            {
+            }
+
+            public override TestResult Execute(TestExecutionContext context)
+            {
+                string key = Test.Id;
+                Test2RetryCount[key] = -1;
```
Comment (dgozman) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:56]: - I'd call the map `Test2RunCount` and initialize with zero.
- What do you think about using `context.CurrentTest.Properties` instead?

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+                while (Test2RetryCount[key] < PlaywrightSettingsProvider.Retries)
+                {
+                    try
+                    {
+                        Test2RetryCount[key]++;
```
Comment (dgozman) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:61]: This could be outside of `try/catch`.

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+                        context.CurrentResult = innerCommand.Execute(context);
+                    }
+                    catch (Exception ex)
+                    {
+                        context.CurrentResult = context.CurrentTest.MakeTestResult();
+                        context.CurrentResult.RecordException(ex);
```
Comment (dgozman) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:67]: Why do we have to handle this? I'd think that `innerCommand` should already do proper error reporting?

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+                    }
+                }
+                if (Test2RetryCount[key] > 0)
+                {
+                    if (!string.IsNullOrEmpty(context.CurrentResult.Output))
+                    {
+                        context.CurrentResult.OutWriter.Write("\n");
+                    }
+                    var retryCount = Test2RetryCount[key];
+                    context.CurrentResult.OutWriter.Write($"Test was retried {retryCount} time{(retryCount > 1 ? "s" : "")}.");
```
Comment (dgozman) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:81]: Could you add example output to the PR description?

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+                }
+                Test2RetryCount.Remove(key);
```
Comment (dgozman) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:83]: Why this?

File: src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+        public override MSTestUnitTesting.TestResult[] Execute(MSTestUnitTesting.ITestMethod testMethod)
+        {
+            var key = testMethod.TestClassName + "." + testMethod.TestMethodName;
+            MSTestUnitTesting.TestResult[] results = new MSTestUnitTesting.TestResult[] { };
+            Test2RetryCount[key] = -1;
+            while (Test2RetryCount[key] < PlaywrightSettingsProvider.Retries)
+            {
+                Test2RetryCount[key]++;
+                results = base.Execute(testMethod);
```
Comment (dgozman) [inline src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs:46]: Shouldn't we append to results here, so that line 64 returns all results?

Comment (mxschmitt) [reply src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs:46]: After a discussion with pf we wanted to implemented for now a primitive approach, since if we return a failed test inside that list, it will result with exitCode 1 which we want to prevent.

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs (hunk 12 above)
Comment (mxschmitt) [reply src/Playwright.NUnit/PlaywrightTestAttribute.cs:56]: Much better, good tip!

**Review by nohwnd — APPROVED** [review]

File: src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs [hunk 18]
```diff
@@ -73,6 +73,12 @@ public static string BrowserName
             private set { }
         }
 
+        public static int Retries
+        {
+            get => _settings?.Retries ?? 0;
+            private set { }
```
[the remark this answers is not in the document]
Comment (mxschmitt) [reply src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs]: nice, thought that I'd have to keep it there.

File: src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MSTestUnitTesting = Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Playwright.MSTest
+{
+    public class PlaywrightTestMethodAttribute : MSTestUnitTesting.TestMethodAttribute
+    {
+        internal static TestContext TestContext { get; set; } = null!;
+
+        private static int RetryCount { get => (int)TestContext.Properties["RetryCount"]; }
```
Comment (kblok) [inline src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs:38]:
```suggestion
        private static int RetryCount => (int)TestContext.Properties["RetryCount"];
```

File: src/Playwright.NUnit/PlaywrightTestAttribute.cs [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+                    var retryCount = Test2RetryCount(context.CurrentTest);
+                    context.CurrentResult.OutWriter.Write($"Test was retried {retryCount} time{(retryCount > 1 ? "s" : "")}.");
+                }
+                return context.CurrentResult;
+            }
+
+            private static int Test2RetryCount(Test test)
+            {
+                return (int)test.Properties.Get("RetryCount")!;
+            }
```
Comment (kblok) [inline src/Playwright.NUnit/PlaywrightTestAttribute.cs:89]:
```suggestion
            private static int Test2RetryCount(Test test)
                => (int)test.Properties.Get("RetryCount")!;
```

File: src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs [hunk 21]
```diff
@@ -70,7 +69,11 @@ public static string BrowserName
                 }
                 return null;
             }
-            private set { }
+        }
+
+        public static int Retries
+        {
+            get => _settings?.Retries ?? 0;
         }
```
Comment (kblok) [inline src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs:77]:
```suggestion
        public static int Retries => _settings?.Retries ?? 0;
```

File: src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs (hunk 19 above)
Comment (campersau) [inline src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs:38]: Why not implement a setter? So you can use the property instead of using this:
- `TestContext.Properties["RetryCount"] = 0;` => `RetryCount = 0;`
- `TestContext.Properties["RetryCount"] = RetryCount + 1;` => `RetryCount++;`
Which makes the last one a lot more readable.

File: src/Playwright.TestAdapter/PlaywrightSettingsXml.cs [hunk 22]
```diff
@@ -57,6 +57,10 @@ public PlaywrightSettingsXml(XmlReader reader)
                         reader.Read();
                         ExpectTimeout = float.Parse(reader.Value);
```
Comment (campersau) [inline src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:58]: This will use the `CurrentCulture` is that intentional?

Comment (mxschmitt) [reply src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:58]: seems unrelated to that PR do you mind filing an issue with more details or are you interested in contributing a fix to it?
